lib/ansible/modules/remote_management/dellemc/idrac/dellemc_idrac_firmware.py

```diff
@@ -0,0 +1,228 @@
+        available updates.
+    - Network share should contain a valid repository of Update Packages (DUPs) and a catalog file describing the DUPs.
+    - All applicable updates contained in the repository are applied to the system.
+    - This feature is available only with iDRAC Enterprise License.
+options:
+    idrac_ip:
+        required: True
+        description: iDRAC IP Address.
+    idrac_user:
+        required: True
+        description: iDRAC username.
+    idrac_pwd:
+        required: True
+        description: iDRAC user password.
+    idrac_port:
+        required: False
+        description: iDRAC port.
+        default: 443
+    share_name:
+        required: True
+        description: CIFS or NFS Network share.
+    share_user:
+        required: False
+        description: Network share user in the format 'user@domain' or 'domain\\user' if user is
+            part of a domain else 'user'. This option is mandatory for CIFS Network Share.
+    share_pwd:
+        required: False
+        description: Network share user password. This option is mandatory for CIFS Network Share.
+    share_mnt:
+        required: True
+        description: Local mount path of the network share with read-write permission for ansible user.
+            This option is mandatory for Network Share.
+    reboot:
+        required: False
+        description: Whether to reboots after applying the updates or not.
+        default: False
+        type: bool
+    job_wait:
+        required:  True
+        description: Whether to wait for job completion or not.
+        type: bool
+        default: True
+    catalog_file_name:
+        required: False
+        description: Catalog file name relative to the I(share_name).
+        type: str
+        default: 'Catalog.xml'
+
+requirements:
+    - "omsdk"
+    - "python >= 2.7.5"
+author: "Rajeev Arakkal (@rajeevarakkal)"
+"""
+
+EXAMPLES = """
+---
+- name: Update firmware from repository on a Network Share
+  dellemc_idrac_firmware:
+       idrac_ip:   "{{ idrac_ip }}"
+       idrac_user: "{{ idrac_user }}"
+       idrac_pwd:  "{{ idrac_pwd }}"
+       share_name: "xx.xx.xx.xx:/share"
+       share_user: "{{ share_user }}"
+       share_pwd:  "{{ share_pwd }}"
+       share_mnt: "/mnt/share"
+       reboot:     True
+       job_wait:   True
+       catalog_file_name:  "Catalog.xml"
+"""
+
+RETURN = """
+---
+msg:
+    description: Updates firmware from a repository on a network share (CIFS, NFS).
+    returned: success
+    type: dict
+    sample: {
+        'CompletionTime': '2018-01-24T00:47:33',
+        'Id': 'JID_XXXXXXXXXXXX',
+        'JobState': 'Completed',
+        'JobType': 'ImportConfiguration',
+        'Message': 'Successfully imported and applied Server Configuration Profile.',
+        'MessageId': 'SYSXXX',
+        'Name': 'Import Configuration',
+        'StartTime': 'TIME_NOW',
+        'Status': 'Success',
```

Comment:
This is probably not needed, because `exit_json()` automatically adds success state key to the result and `fail_json()` adds the failure flag there.
Another thing to think about is `changed` flag: when this module changes anything on the target it should be `True`.

Reply:
this is part of the native return value of the firmware update job details which we are not modifying with in the module.

Reply:
k